### PR TITLE
done homework

### PR DESCRIPTION
**Pull request by Mohammad1704**
### How difficult did you find this (0 - 10)? (0 very easy, 10 very difficult)
7 

### How long did it take?
2h

### Was there anything that you struggled with?  If so, what?
hash !  I don't understand it very will

### Is there anything that you would like some further information on?
no

### Do you have any suggestions to improve this assignment?

no

### CODE REVIEW

File: data-collections-exercises.md [hunk 1]
```diff
@@ -3,95 +3,96 @@
...
 ```rb
-# Your answer here
+#  planeteers[1]
 ```
 
 Add "Heart" to the end of `planeteers`.
 
 ```rb
-# Your answer here
+#  planeteers.collect {|p| p + " Heart" }
```
Comment (ghadeer-x) [inline data-collections-exercises.md:18]: what a way of solving this problem !! 👏 , in this way you will add `Heart` for every element in the array so the array will be something like this 

```
["Earth Heart", "Wind Heart", "Captain Planet Heart", "Fire Heart", "Water Heart"] 
```

but the goal is to add another element at the end of the array, you can use the `push` method to achieve that 
`planeteers.push("Heart")`

File: data-collections-exercises.md [hunk 2]
```diff
@@ -3,95 +3,96 @@
...
 ```
 
 Remove "Captain Planet" from the array (without using a method).
 
 ```rb
-# Your answer here
+# planeteers.delete("Captain Planet")
```
Comment (ghadeer-x) [inline data-collections-exercises.md:24]: you used a method but you solved the problem you can use `-` operator to delete an element without using a method `planeteers - ["Captain Planet"]`

File: data-collections-exercises.md [hunk 3]
```diff
@@ -3,95 +3,96 @@
...
 ```rb
-# Your answer here
+# ninja_turtle[:age] = 12
 ```
 
 Remove the `radical` key-value pair from `ninja_turtle`.
 
 ```rb
-# Your answer here
+# ninja_turtle.reject { |k,v| k == :radical }
```
Comment (ghadeer-x) [inline data-collections-exercises.md:77]: using this way to solve you won't be able to delete radical directly from ninja_turtle instead you will generate a copy of ninja_turtle hash without the radical key to delete it directly from the hash we use the delete method `ninja_turtle.delete:radical`

**Comment by ghadeer-x** [comment]
Great job Mohammad 👍,  hope you manage to fully understand Hashes by now 💯

File: data-collections-exercises.md (hunk 2 above)
Comment (Mohammad1704) [reply data-collections-exercises.md:24]: oh!
sorry I mixed up between function and method in hash and react

File: data-collections-exercises.md (hunk 3 above)
Comment (Mohammad1704) [reply data-collections-exercises.md:77]: yeah, you right.
it rejects a copy of ninja_turtle without radical key
I thought it reject radical key form the hash and remove it to different value or hash

**Pull request closed without merging**